RELEASING.md

```diff
@@ -5,8 +5,8 @@
 1. Make sure all relevant changes for this release are included under `Unreleased` section in `CHANGELOG.md` and are in language that non-contributors to the project can understand.
 
```

Comment:
Can we add one more point to the checklist: we should also check if any of the changes made since the last release broke ABI compatibility. If so, [this number](https://github.com/open-telemetry/opentelemetry-cpp/blob/master/api/include/opentelemetry/version.h#L5) should be updated.

Reply:
Good point. Have updated this information. Also we should be updating major version ( in release tag) for any ABI/API change. I have added guidelines for that too in `CHANGELOG.md` ( slightly modified from semver guidelines to indicate ABI change)